File: demo/src/main/java/com/example/demo/model/Entity/Note.java
package com.example.demo.model.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
public class Note {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String content;
    private boolean archived = false;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: frontend/src/models/Note.ts
import { Category } from './category';

export interface Note {
  id: number;
  title: string;
  content: string;
  archived: boolean;
  category: Category;
}

export interface NoteReqDTO {
  title: string;
  content?: string;
  archived: boolean;
  categoryId: number;
}